### PR TITLE
Use redis datastore

### PR DESCRIPTION
**Pull request by digitalronin**
Changes to support optionally using redis as the datastore, and deployment to Heroku.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -72,10 +77,19 @@ console.log('config', config);
...
-var smtp = nodemailer.createTransport('SMTP', config.mailTransportConfig);
+var smtp = nodemailer.createTransport({
+  host: process.env.MAILGUN_SMTP_SERVER,
+  port: process.env.MAILGUN_SMTP_PORT,
+  secure: false,
+  auth: {
+    user: process.env.MAILGUN_SMTP_LOGIN,
+    pass: process.env.MAILGUN_SMTP_PASSWORD
+  }
+});
```
Comment (hanshuebner) [inline server.js:88]: Can you make the mailer configuration conditional on the presence of `config.mailTransportConfig` and the `MAILGUN.*` configuration variables?  Something like
```
var smtp;
if (config.mailTransportConfig) {
  smtp = nodemailer.createTransport('SMTP', config.mailTransportConfig);
} else if (env.MAILGUN_SMTP_SERVER) {
  smtp = nodemailer.createTransport({
    host: process.env.MAILGUN_SMTP_SERVER,
    port: process.env.MAILGUN_SMTP_PORT,
    secure: false,
    auth: {
      user: process.env.MAILGUN_SMTP_LOGIN,
      pass: process.env.MAILGUN_SMTP_PASSWORD
    }
  });
} else {
  console.log('email sending not configured');
}
```

File: server.js [hunk 2]
```diff
@@ -176,12 +191,15 @@ Game.prototype.makeLink = function(player)
 Game.prototype.sendInvitation = function(player, subject)
 {
     try {
+        const gameLink = this.makeLink(player)
+        const sender = `scrabble@${process.env.MAILGUN_DOMAIN}`;
```
Comment (hanshuebner) [inline server.js:195]: Maybe processing the environment configuration into the config object upon startup would be nicer than spreading the conditionals over the code.  This is another place where such a conditional would be needed.

File: server.js [hunk 3]
```diff
@@ -535,8 +553,6 @@ Game.prototype.finish = function(reason) {
                        })
                      };
     game.endMessage = endMessage;
-
-    db.snapshot();
```
Comment (hanshuebner) [inline server.js:539]: Please add an empty `snapshot()` function to `redis.js` so that the functionality is retained for `node-dirty`.

**Comment by digitalronin** [comment]
How's this?

**Comment by hanshuebner** [comment]
Awesome, thanks!

**Pull request merged**